### PR TITLE
Fix and unit tests in models

### PR DESCRIPTION
**Pull request by southfox**

### CODE REVIEW

File: Example/Forescoop Extension/InterfaceController.swift [hunk 1]
```diff
@@ -52,8 +53,8 @@ private extension InterfaceController {
     
     func updateForecast() {
         Task { [weak self] in
-            guard let spotId = try? await ForecastWindguruService.instance.searchSpots(byLocation: "Bariloche")?.firstSpot?.id else { throw CustomError.cannotFindSpotId }
-            let spotForecast = try? await ForecastWindguruService.instance.forecast(bySpotId: spotId)
+            guard let spotId = try? await forecastService?.searchSpots(byLocation: "Bariloche")?.firstSpot?.id else { throw CustomError.cannotFindSpotId }
```
Comment (southfox) [inline Example/Forescoop Extension/InterfaceController.swift:56]: using Dependency injection, no need to use singletons here.

File: Example/Forescoop/MainViewController.swift [hunk 2]
```diff
@@ -30,9 +30,24 @@ class MainViewController: UIViewController {
     @IBOutlet weak var loginButton: UIButton!
     
     var passwordTextField: UITextField?
+    var forecastService: ForecastWindguruProtocol? = ForecastWindguruService()
     
     var user: User?
     
+    init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?, forecastService: ForecastWindguruProtocol? = nil) {
```
Comment (southfox) [inline Example/Forescoop/MainViewController.swift:37]: Dependency injection here.

File: Example/Tests/MainViewControllerTests.swift [hunk 3]
```diff
@@ -7,18 +7,19 @@
...
 final class MainViewControllerTests: XCTestCase {
 
-    private var vc: MainViewController!
-    
+    private var vc: MainViewController?
     override func setUpWithError() throws {
         let sb = UIStoryboard(name: "Main", bundle: nil)
         vc = sb.instantiateViewController(identifier: "MainViewController") { coder in
-            MainViewController(coder: coder)
+            let forecastService = ForecastWindguruMockup()
```
Comment (southfox) [inline Example/Tests/MainViewControllerTests.swift:19]: Using a mockup for testing.

File: Forescoop/Classes/Definition.swift [hunk 4]
```diff
@@ -22,4 +53,49 @@ public struct Definition {
...
+            let dict = NSDictionary.init(contentsOfFile: path) as? [String: AnyObject] else {
+                return nil
+        }
+        return dict
+    }()
+}
+
+private extension Definition {
+
+    var bundleFromFramework: Bundle? {
```
Comment (southfox) [inline Forescoop/Classes/Definition.swift:77]: Probably need to move this to another place.

File: Forescoop/Classes/ForecastWindguruMockup.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
+//
+//  ForecastWindguruMockup.swift
+//  Forescoop
+//
+//  Created by fox on 07/06/2023.
+//
+
+import Foundation
```
Comment (southfox) [inline Forescoop/Classes/ForecastWindguruMockup.swift:8]: here the mockup, need to implement this to use the json files that are already in the unit tests.

**Review by jhf2021 — APPROVED** [review]

**Pull request merged**